src/streaming/pipeline/server.rs

```diff
@@ -131,8 +141,28 @@ impl<P, T, B, S> super::advanced::Dispatch for Dispatch<S, T, P> where
     }
 
     fn poll(&mut self) -> Poll<Option<PipelineMessage<Self::In, Self::Stream, Self::Error>>, io::Error> {
-        for slot in self.in_flight.iter_mut() {
-            slot.poll();
+        // new Futures must be polled once to start processing
+        if self.polled_to < self.in_flight.len() {
```

Comment:
Yes, could be easily moved to fn dispatch, a few lines up in the same file.
I just wasn't sure if that would break any non-obvious assumption, so I preferred to keep the poll at the same place as before, for now.